src/inference/include/openvino/runtime/config.hpp

```diff
@@ -0,0 +1,380 @@
+ *
+ * Metric returns a value of std::vector<uint32_t> type, where:
+ *  - First value is bottom bound.
+ *  - Second value is upper bound.
+ */
+static constexpr Key<std::tuple<unsigned int, unsigned int>> range_for_streams{"RANGE_FOR_STREAMS"};
+
+/**
+ * @brief Metric to provide a hint for a range for number of async infer requests. If device supports streams,
+ * the metric provides range for number of IRs per stream.
+ *
+ * Metric returns a value of std::vector<uint32_t> type, where:
+ *  - First value is bottom bound.
+ *  - Second value is upper bound.
+ *  - Third value is step inside this range.
+ */
+static constexpr Key<std::tuple<unsigned int, unsigned int, unsigned int>> range_for_async_infer_requests{
+    "RANGE_FOR_ASYNC_INFER_REQUESTS"};
+
+/**
+ * @brief Metric which defines the device architecture.
+ */
+static constexpr Key<std::string> device_architecture{"DEVICE_ARCHITECTURE"};
+
+namespace runtime {
+
+/**
+ * @brief Enum to define possible device types
+ */
+enum class DeviceType {
+    integrated = 0,
+    discrete = 1,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const DeviceType& deviceType) {
+    switch (deviceType) {
+    case DeviceType::discrete:
+        return os << "discrete";
+    case DeviceType::integrated:
+        return os << "integrated";
+    default:
+        return os << "unknown";
+    }
+}
+/** @endcond */
+
+}  // namespace runtime
+
+/**
+ * @brief Metric to get a type of device. See DeviceType enum definition for possible return values
+ */
+static constexpr Key<::ov::runtime::DeviceType> device_type{"DEVICE_TYPE"};
+
+/**
+ * @brief Metric which defines Giga OPS per second count (GFLOPS or GIOPS) for a set of precisions supported by
+ * specified device
+ */
+static constexpr Key<std::map<ie::Precision, float>> device_gops{"DEVICE_GOPS"};
+
+/**
+ * @brief Metric which defines support of import/export functionality by plugin
+ */
+static constexpr Key<bool> import_export_support{"IMPORT_EXPORT_SUPPORT"};
+
+/**
+ * @brief Metric to get a name of network
+ */
+static constexpr Key<std::string> network_name{"NETWORK_NAME"};
+
+/**
+ * @brief  Metric to get a float of device thermal
+ */
+static constexpr Key<float> device_thermal{"DEVICE_THERMAL"};
+
+/**
+ * @brief Metric to get an unsigned integer value of optimal number of executable network infer requests.
+ */
+static constexpr Key<uint32_t> optimal_number_of_infer_requests{"OPTIMAL_NUMBER_OF_INFER_REQUESTS"};
+
+/**
+ * @brief High-level OpenVINO Performance Hints
+ * unlike low-level config keys that are individual (per-device), the hints are somthing that every device accepts
+ * and turns into device-specific settings
+ */
+static constexpr Key<std::string> performance_hint{"PERFORMANCE_HINT"};
+static constexpr auto latency = "LATENCY";        //!< Latency hint
+static constexpr auto throughput = "THROUGHPUT";  //!< Throughput hint
```

Comment:
Why not an enum?

Reply:
To preserve backward compatibility with plugins.

Reply:
Can we handle this with `Any` class? For example:

```
enum class PerformanceHint { ... };

// in. cpp file
EnumNames<PerformanceHint>& EnumNames<PerformanceHint>::get() { ... }

Any val = PerformanceHint::LATENCY;
assert(val.as<PerformanceHint> == PerformanceHint::LATENCY);
assert(val.as<std::string> == "LATENCY");
```

Reply:
Any already has an `Any::print(std::ostream&)` based on stored type support of `operator<<`. And current types are already converted to strings before passing to plugins.
So you are right we can replace it with enums.

But what about scalability? 
What if some plugin supports custom performance hints? For example, the plugin provides the metric "SUPPORTED_PERFOMANCE_HINTS" with the list of custom performance hints

Reply:
Replaced with enum